### ISSUE TITLE
EMIS deprecation: migrate emis veteran_status

### ISSUE THREAD

**Issue by LindseySaari** [department-of-veterans-affairs/va.gov-team#44856]
## Description
The external service EMIS is being deprecated. EMIS is an external service integrated into vets-api and appears to be heavily engrained into the vets-api codebase. 

There are many random places in the main app where EMIS is utilized for `veteran_status`. We should be sure that `veteran_status` is no longer pulled from EMIS. See Main App Responsibilities ([namely the veteran? and military_person? methods](https://vfs.atlassian.net/wiki/spaces/BTT/pages/2264694838/EMIS+Deprecation#Main-App-Responsibilities)). The User model utilizes EMIS via [delegation methods](https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/models/user.rb#L326-L327). There’s some [metaprogramming](https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/models/user.rb#L406-L416) used to initialize the methods used to obtain the EMIS related information

Some of the underlying base methods may have the ability to be reused, but we will need to be sure that `veteran_status` is no longer being pulled from EMIS. We should consider this a breaking change.

All calls to EMIS should be removed and code should be migrated to use VaProfile. We can most likely use the `https://qa.vaprofile.va.gov:7005/military-personnel/swagger-ui/index.html?configUrl=/military-personnel/v3/api-docs/swagger-config` endpoint from the [swagger docs](https://qa.vaprofile.va.gov:7005/military-personnel/swagger-ui/index.html?configUrl=/military-personnel/v3/api-docs/swagger-config)

Also see:
1. [EMIS deprecation discovery](https://vfs.atlassian.net/wiki/spaces/BTT/pages/2264694838/EMIS+Deprecation)
2. [Prioritization](https://vfs.atlassian.net/wiki/spaces/BTT/pages/2274426885/EMIS+deprecation+prioritization)
3. [EMIS pre-fill impact](https://vfs.atlassian.net/wiki/spaces/BTT/pages/2286714883/EMIS+Pre-fill+Impact+WIP)

Notes:
1. Id card attributes and controller are two examples of where veteran_status is used. [See details in table](https://vfs.atlassian.net/wiki/spaces/BTT/pages/2264694838/EMIS+Deprecation#Main-App-Responsibilities)

## Acceptance Criteria
- [x] Investigate if we can use anything that the auth experience team has already done
- [x] Migrate veteran_status to VaProfile
- [x] Tests are passing
- [ ] Update documentation
- [x] Generate Additional Tickets as necessary
- [x] implement caching (based on findings in https://github.com/department-of-veterans-affairs/va.gov-team/issues/64549)

**Comment by LindseySaari** [comment]
The current veteran_status returns [these attributes](https://github.com/department-of-veterans-affairs/vets-api/blob/e934b2a4ec7b0025027d5124ca3337d5032fea2e/app/models/emis_redis/veteran_status.rb#L66-L79)

**Comment by laineymajor** [comment]
This is a sep service class. Let's explore taking this on now as we need to get this done to unblock VFS teams.

**Comment by laineymajor** [comment]
For Ryan to work

**Comment by LindseySaari** [comment]
It looks like for the veteran_status info, we can still use the sameprofile-service endpoint but a slightly different bio. It would be the `militaryPerson.militarySummary` bio

```        
"title38StatusText": "string",
"title38StatusCode": "string",
```

**Comment by rjohnson2011** [comment]
Built out new VeteranStatus model, configuration, service, and response - https://github.com/department-of-veterans-affairs/vets-api/commits/v1_va_profile_veteran_status

Currently touching base with Lindsey several times throughout the day to chat and continue making progress. Incorporating passing specs as the primary remaining portion of this ticket.

**Comment by laineymajor** [comment]
Continuing out the spec portion.
Mild blocker with getting some of the models connected (had a failing spec).... now past the original failing spec.

Currently working in the veteran_status spec.

**Comment by laineymajor** [comment]
@rjohnson2011 will you be able to wrap this up today? This is blocking other work, so we'd like it wrapped up before you move on to your other work next week. Thanks!

**Comment by laineymajor** [comment]
This will carryover to next sprint. Near complete. @rjohnson2011 for additional details.

**Comment by laineymajor** [comment]
lindsey and ryan met this morning. recorded a couple of cassettes for the non-happy path testing. happy path testing is complete.

**Comment by laineymajor** [comment]
ryan will work on the caching ticket once this is complete.

**Comment by laineymajor** [comment]
teamed with lindsey and rebecca to review errors. will continue to collaborate re errors.

**Comment by LindseySaari** [comment]
Tests are passing now. Ryan is working on wrapping the calls in a feature toggle to move onto testing in lower envs

**Comment by LindseySaari** [comment]
Rachal is starting to look into the caching and if we would benefit from a caching strategy

**Comment by LindseySaari** [comment]
Tests are passing and the next step is to wrap this in a feature flag and test in a live (lower) environment. The testing phase will continue into next sprint, going to tag this ticket with the next sprint

**Comment by LindseySaari** [comment]
PR is open, there were some comments from the identity team. Working through the next testing steps.

**Comment by laineymajor** [comment]
Additional tests were complete yesterday.
Remaining tests in the service spec (3 failing).

**Comment by LindseySaari** [comment]
Tests are green and passing!

**Comment by laineymajor** [comment]
- [ ] testing on review instances
- [ ] once tested, caching
- [ ] hand off to VFS teams for testing

**Comment by laineymajor** [comment]
likely carry into next sprint... for a few days.

**Comment by LindseySaari** [comment]
- [x] Test on review instance
- [ ] Test in dev wrapped in feature flag
- [ ] Caching was moved to another ticket

**Comment by laineymajor** [comment]
Removing this from sprint as it's more of a wait and see to close...
